### ISSUE TITLE
Accessibility Stuff

### ISSUE THREAD

**Issue by KonnorRogers** [josefarias/hotwire_combobox#25]
First off, you've obviously put quite a bit of thought and detail into making this. For the most part these components look solid and you have a put a ton of work into this and I'm not here to bash anything, just pointing out initial things that jump out at me!

- I notice you have the following HTML in your listbox:

```html
<ul role="listbox">
  <div>
    <li></li>
    <li></li>
  </div>
</ul>
```

This generally breaks accessibility tools because `<div>` is not a valid child of a `<ul>`

When run through Voiceover, this seems to cause the combobox to not announce what number option youre on (3 of 10 for example) and it doesn't read when an element is selected.

I also don't see an `aria-activedescendant`  on the `<input>` element. The `aria-activedescendant` should point to the currently active `role="option"`

These are the big things so far. Happy to take another look once these two things have been fixed. The screen reader just kinda gives up and just reads elements but gives me no contextual information like the one found herE:

https://www.w3.org/WAI/ARIA/apg/patterns/combobox/examples/combobox-autocomplete-both/

Finally, the `autocomplete` in Firefox for the Youtube example looks not so hot with both the native browser autocomplete and the combobox options 

<img width="708" alt="Screenshot 2024-02-09 at 1 03 46 PM" src="https://github.com/josefarias/hotwire_combobox/assets/26425882/b3aa4d0e-09d6-4da1-98d1-c61ab724360c">

**Comment by josefarias** [comment]
Really appreciate you taking a look! Not bashing at all. Nothing in the library is unchangeable and I'll be glad to look for ways to address these.

Activedescendant should be a pretty easy fix. The nested div is a hacky solution I spiked into place. I'm happy to revisit.

Thanks again for raising these. I won't be available for a few weeks but will address when I'm back 😄

**Comment by josefarias** [comment]
Hey @KonnorRogers thanks again for pointing these out. I've addressed as follows:

* Non `<li>` elements: addressed in https://github.com/josefarias/hotwire_combobox/pull/55
* aria-activedescendant: addressed in https://github.com/josefarias/hotwire_combobox/pull/56
* Funky browser autocomplete: addressed in https://github.com/josefarias/hotwire_combobox/pull/57

If you're still up for taking another look, the latest stuff has been released in v0.1.37, which is also live at https://hotwirecombobox.com/

Closing this for now. But happy to reopen if you find any regressions or more areas of improvement. Can also do that in another issue, up to you.

Thanks again! The library is much better off after wrapping up this work you helpfully pointed out.
